### DISCUSSION TITLE
Help on Ubuntu install

### DISCUSSION THREAD

**Discussion by marksmaynard1** [thirtythreeforty/neolink#145] [Q&A]
Hi, im fairly new to linux and just swapped my server over from windows. neolink was working fine on windows so i know the issue isnt the camera or config details. i followed the documentation and even followed the guide here to no avail. https://github.com/thirtythreeforty/neolink/discussions/126.
i got the neolink folder in the /usr/local/bin i chmod +x neolink and get my config.toml in the same folder but when i enter ./neolink --config my_config.toml i get 
./neolink: error while loading shared libraries: libgstrtspserver-1.0.so.0: cannot open shared object file: No such file or directory
root@UbuntuServer:/usr/local/bin/neolink# 
i am pulling my hair out and if i cant get this figured out it basically makes the camera useless any help on this would be greatly appreciated

**Comment by QuantumEntangledAndy** [comment]
Ah your missing the dependencies. If you look in the file `docs/unix_setup.md` the details are there but to summarise you need these:

```bash
sudo apt install \
  libgstrtspserver-1.0-0 \
  libgstreamer1.0-0 \
  libgstreamer-plugins-bad1.0-0 \
  gstreamer1.0-plugins-good \
  gstreamer1.0-plugins-bad
```

**Reply by QuantumEntangledAndy** [reply]
In general on Linux if you ever get something of the form:

```
error while loading shared libraries: libgstrtspserver-1.0.so.0: cannot open shared object 
```

Your missing the dependencies. You can usually get them by installing the package for the dependency which will usually have the same name as as in the error but without the version number.

```
apt install libgstrtspserver
```

And if that fails read the docs of the thing your installing and it will list the packages you need somewhere.

**Reply by marksmaynard1** [reply]
you absolute legend lol, im quite a noob in linux ive migrated my whole server plus 20+ programs and ive done alright but this had me pulling my hair out seems to be working now. thank you so much for saving me hours of stress haha

**Reply by QuantumEntangledAndy** [reply]
Get any other issues give me shout :)
